### PR TITLE
trust role support

### PR DESCRIPTION
**Pull request by ranga543**
* Added `createIamTrustRole` function to create `IAM Trust Roles`.
* Updated `README.md`

### CODE REVIEW

**Comment by brianantonelli** [comment]
Thanks! I'll look into this and approve in a bit.

File: lib/alks-api.js [hunk 1]
```diff
@@ -231,6 +231,53 @@ exports.createIamRole = function(account, password, roleName, roleType, includeD
...
+        roleType: roleType,
+        trustArn: trustArn
+    }, account),
+        options = _.extend({
+            debug: false,
+            ua:    DEFAULT_UA
+    }, opts),
+        endpoint = account.server + '/createNonServiceRole/';
+
+    log('api:createNonServiceRole', 'creating IAM trust role at endpoint: ' + endpoint, options);
```
Comment (brianantonelli) [inline lib/alks-api.js]: can you change these two log statements to be `api: createIamTrustRole`?

**Comment by brianantonelli** [comment]
thanks! 🌮 👍

**Comment by ranga543** [comment]
@brianantonelli Yes, I already made the changes for `CLI` just waiting on `alks-node NPM` package to be published.
For TF provider I might need to make some changes to `alks-go` repo. I opened up 1 issue on `ALKS` service [here](https://ghe.coxautoinc.com/ETS-CloudAutomation/ALKS/issues/211) looks like `role` and `account` are required properties for `createNonServiceRole` rest endpoint.

**Pull request merged**